tests/test_bucket_list_items.py

```diff
@@ -0,0 +1,131 @@
+        url = 'api/v1/bucketlist/1/bucketlistitem'
+        response = self.client.post(url, data=json.dumps(dict(item_name='mocking')),\
+                                                headers=self.set_header())
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['name'] == 'mocking')
+        url = 'api/v1/bucketlist/1/bucketlistitem/1'
+        response = self.client.put(url, data=json.dumps(dict(item_name='functional testing')),\
+                                                headers=self.set_header())
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['name'] == 'functional testing')
+
+    def test_update_non_existant_item(self):
+        """
+         test fxn update if it updates non existant bucketlist item
+        """
+
+        response = self.create_bucketlist()
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['name'] == 'test')
+
+        url = 'api/v1/bucketlist/1/bucketlistitem/1'
+        response = self.client.put(url, data=json.dumps(dict(item_name='functional testing')),\
+                                                headers=self.set_header())
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['error'] == 'bucket list item does not exist')
+
+    def test_delete_item(self):
+        """
+        test  delete bucketlist item
+        """
+
+        response = self.create_bucketlist()
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['name'] == 'test')
+
+        url = 'api/v1/bucketlist/1/bucketlistitem'
+        response = self.client.post(url, data=json.dumps(dict(item_name='mocking')),\
+                                                headers=self.set_header())
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['name'] == 'mocking')
+
+        url = 'api/v1/bucketlist/1/bucketlistitem/1'
+        response = self.client.delete(url, headers=self.set_header())
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['message'] == 'bucket list item deleted')
+
+    def test_delete_non_existing_item(self):
+        """
+        test if can delete a non existant bucket list item
+        """
+
+        response = self.create_bucketlist()
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['name'] == 'test')
+
+        url = 'api/v1/bucketlist/1/bucketlistitem'
+        response = self.client.post(url, data=json.dumps(dict(item_name='mocking')),\
+                                                headers=self.set_header())
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['name'] == 'mocking')
+
+        url = 'api/v1/bucketlist/1/bucketlistitem/2'
+        response = self.client.delete(url, headers=self.set_header())
+        data = json.loads(response.data.decode())
+        self.assertTrue(data['message'] == 'bucket list item does not exist')
+
+        
```

Comment:
Most test cases you are repeating yourself. Like you are testing again if a bucketlist is created. That is not relevant since you've tested it already.